build/Scripts/PrivateXdt/XdtHelper.psm1

```diff
@@ -32,19 +32,38 @@ function Get-EnvironmentVariableAddInfo {
         [Parameter(Mandatory=$true)]
         [ValidateNotNullOrEmpty()]
         [string]
-        $Transform
+        $Transform,
+
+        [Parameter(Mandatory=$false)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $Value
     )
 
     # Operation affects the placement of this element.
     # <add name="$EnvironmentVariable" xdt:Locator="Match(name)" xdt:Transform="$Transform" />
-    Write-Output @{
-        destination = '/configuration/system.webServer/runtime/environmentVariables'
-        operation = $Operation
-        element = 'add'
-        attributes = [ordered] @{
-            name = $EnvironmentVariable
-            'xdt:Locator' = 'Match(name)'
-            'xdt:Transform' = $Transform
+    if ($Value) {
```

Comment:
Rather than duplicating the object creation, why not create the object and then conditionally set the value property on the attributes object?